crates/storage/src/blueprint.rs

```diff
@@ -75,6 +76,32 @@ where
             })
             .transpose()
     }
+
+    /// Returns multiple values from the storage.
+    fn get_batch<'a, Iter>(
+        storage: &'a S,
+        keys: Iter,
+        column: S::Column,
+    ) -> impl Iterator<Item = crate::Result<Option<M::OwnedValue>>> + 'a
```

Comment:
nit: 
```suggestion
    ) -> impl IntoIterator<Item = crate::Result<Option<M::OwnedValue>>> + 'a
```

`Iterator`s implement `IntoIterator`, so if you change the signature to return a `IntoIterator` then `StorageInspect` becomes slightly more flexible, as implementations can return either an iterator or a collection.

Reply:
As a general principle, I like interfaces to be very general in what they accept but prefer them to be specific in what they return. (see [Robusness principle](https://en.wikipedia.org/wiki/Robustness_principle)). That's why I typically accept `IntoIterator` in parameters, because as you say then the caller doesn't have to care about doing `.into_iter()´ explicitly if they have a collection or anything.

Therefore, for return values it's better in my opinion to return the `Iterator` directly since it is more specific. Otherwise, all callers are basically forced to call `.into_iter()` before they can do anything with the returned iterator, which is less ergonomic. Any implementer of the trait can (and should) always call `.into_iter()` if they are about to return a collection. Again, not doing so would just force this burden upon the user of the function which is less than ideal without providing any benefit whatsoever.

Let me know if you disagree and I'll be happy to further discuss this.